test/utils/TestUtils.ts

```diff
@@ -270,164 +272,175 @@ const deployRelayHub = async (
-    envelopingRequest,
-    owner
-  );
-
-  const transaction = await factory
-    .connect(sender)
-    .relayedUserSmartWalletCreation(
-      envelopingRequest.request as DeployRequestBody,
-      suffixData,
+    const signature = signData(dataTypesToSign, valuesToSign, owner);
+    await (factory as CustomSmartWalletFactory).createUserSmartWallet(
+      owner.address,
+      constants.AddressZero,
       constants.AddressZero,
+      index,
+      '0x',
       signature
     );
 
-  const receipt = await transaction.wait();
-
-  console.log('Cost of deploying SmartWallet: ', receipt.cumulativeGasUsed);
-
-  const isCustom =
-    isCustomSmartWallet ?? (!!logicAddr && logicAddr !== constants.AddressZero);
-
-  const swAddress = isCustom
-    ? await (factory as CustomSmartWalletFactory).getSmartWalletAddress(
-        owner.address,
-        recoverer,
-        logicAddr,
-        initParams,
-        index
-      )
-    : await (factory as SmartWalletFactory).getSmartWalletAddress(
-        owner.address,
-        recoverer,
-        index
-      );
-
-  return isCustom
-    ? ethers.getContractAt('CustomSmartWallet', swAddress)
-    : ethers.getContractAt('SmartWallet', swAddress);
-};
+    const customSmartWalletAddress = await (
+      factory as CustomSmartWalletFactory
+    ).getSmartWalletAddress(
+      owner.address,
+      constants.AddressZero,
+      constants.AddressZero,
+      keccak256('0x'),
+      index
+    );
 
-const prepareRelayTransaction = async ({
-  relayHub,
-  owner,
-  tokenContract = constants.AddressZero,
-  tokenAmount = 0,
-  tokenGas = 0,
-  validUntilTime = 0,
-  logicAddr = constants.AddressZero,
-  initParams = '0x00',
-  gas = 0,
-  swAddress,
-}: PrepareRelayTransactionParams) => {
-  const envelopingRequest = createEnvelopingRequest(
-    true,
-    {
-      relayHub,
-      from: owner.address,
-      to: logicAddr,
-      data: initParams,
-      tokenContract,
-      tokenAmount,
-      tokenGas,
-      validUntilTime,
-      gas,
-    },
-    {
-      callForwarder: swAddress,
-    }
-  );
+    return await ethers.getContractAt(
+      'CustomSmartWallet',
+      customSmartWalletAddress
+    );
+  } else {
+    const dataTypesToSign = ['bytes2', 'address', 'address', 'uint256'];
+    const valuesToSign = [
+      '0x1910',
+      owner.address,
+      constants.AddressZero,
+      index,
+    ];
 
-  const { signature } = await signEnvelopingRequest(envelopingRequest, owner);
+    const signature = signData(dataTypesToSign, valuesToSign, owner);
+    await (factory as SmartWalletFactory).createUserSmartWallet(
+      owner.address,
+      constants.AddressZero,
+      index,
+      signature
+    );
 
-  return {
-    envelopingRequest,
-    signature,
-  };
+    const smartWalletAddress = await (
+      factory as SmartWalletFactory
+    ).getSmartWalletAddress(owner.address, constants.AddressZero, index);
+
+    return await ethers.getContractAt('SmartWallet', smartWalletAddress);
```

Comment:
what about if its a `CustomSmartWallet`.

Reply:
Notice that this is the else fo an if-else block

Reply:
you are right, sorry :S